resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/KubernetesSuite.scala

```diff
@@ -306,8 +305,16 @@ class KubernetesSuite extends SparkFunSuite
         }
       })
 
-    val patienceInterval = interval.getOrElse(INTERVAL)
-    Eventually.eventually(TIMEOUT, patienceInterval) { execPods.values.nonEmpty should be (true) }
+    val (patienceInterval, patienceTimeout) = {
+      executorPatience match {
+        case Some(patience) => (patience._1.getOrElse(INTERVAL), patience._2.getOrElse(TIMEOUT))
```

Comment:
Can you pattern match a tuple inside Some? I forget if there's a reason you can't

Reply:
It does not work with None in the tuple also pattern matching on None is[ less idiomatic](https://www.scala-lang.org/api/2.9.0/scala/Option.html).
I could even refactor to:
```
    executorPatience.map(p => (p._1.getOrElse(INTERVAL), p._2.getOrElse(TIMEOUT)))
      .getOrElse((INTERVAL, TIMEOUT))
```